voluncheer/profiles/test_forms.py

```diff
@@ -0,0 +1,108 @@
+from django.test import TestCase
+
+from profiles.forms.volunteers import VolunteerCreationForm
+from profiles.forms.organizations import OrganizationCreationForm
+
+
+class OrganizationCreationFormTest(TestCase):
+    def setUp(self):
+        """See base class."""
+        self.form = OrganizationCreationForm()
+
+    def test_name_label(self):
+        """Test name field is labeled correctly"""
+        self.assertTrue(
+            self.form.fields["name"].label is None
+            or self.form.fields["name"].label == "Name"
+        )
+
+    def test_organization_must_submit_unique_email(self):
+        """Test that organization email must be unique"""
+        data1 = {
+            "name": "Sith Lords",
+            "email": "sith@sith.com",
+            "password1": "come_to_the_dark_side",
+            "password2": "come_to_the_dark_side",
+            "type": "UserType.ORGANIZATION,",
+        }
+        data2 = {
+            "name": "The Sith Lords",
+            "email": "sith@sith.com",
+            "password1": "come_to_the_dark_side_again",
+            "password2": "come_to_the_dark_side_again",
+            "type": "UserType.ORGANIZATION,",
+        }
+
+        form1 = OrganizationCreationForm(data=data1)
+        self.assertTrue(form1.is_valid())
+        form1.save(self)  # Submitting form 1
+        form2 = OrganizationCreationForm(data=data2)
+        self.assertFalse(form2.is_valid())
+
+
+class VolunteerCreationFormTest(TestCase):
+    def setUp(self):
+        """See base class."""
+        self.form = VolunteerCreationForm()
+
+    def test_first_name_label(self):
+        """Test first name field is labeled correctly"""
+        self.assertTrue(
+            self.form.fields["first_name"].label is None
```

Comment:
Why do we need to check if the label is `None` OR the expected value?

Reply:
According to the tutorial I followed:

"Note here that we also have to test whether the label value is None, because even though Django will render the correct label it returns None if the value is not explicitly set."

I agree it's a bit strange to test for the default None value since the labels are not explicitly set in any of the models, but I suppose it protects us if we someday decide to add labels and they are entered incorrectly. Also happy to remove the tests if we think they are just fluff.

Reply:
I don't like it but if that's what Django recommends we should stick with it. Thanks for the explanation.